test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java

```diff
@@ -120,45 +128,120 @@ public String getID() {
-                                                                                      .put("debugLogging", true)
-                                                                                      .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
-                                                                                      //.put("options", props.getProperties())
-                                                                                      .put("options", props)
-                                                                                      .put("image",
-                                                                                       getImageSpec(DOCKER_REGISTRY + PREFIX + "/" + PRAVEGA_IMAGE_NAME, PRAVEGA_VERSION))
-                                                                                      .put("tier2", tier2Spec())
-                                                                                      .build();
+                .put("segmentStoreReplicas", segmentStoreCount)
+                .put("debugLogging", true)
+                .put("cacheVolumeClaimTemplate", pravegaPersistentVolumeSpec)
+                .put("controllerResources", getResources("2000m", "3Gi", "1000m", "1Gi"))
+                .put("segmentStoreResources", getResources("2000m", "5Gi", "1000m", "3Gi"))
+                .put("options", props)
+                .put("image", pravegaImgSpec)
+                .put("tier2", tier2Spec())
+                .build();
+
         return ImmutableMap.<String, Object>builder()
-                .put("apiVersion", "pravega.pravega.io/v1alpha1")
+                .put("apiVersion", CUSTOM_RESOURCE_API_VERSION)
                 .put("kind", CUSTOM_RESOURCE_KIND_PRAVEGA)
                 .put("metadata", ImmutableMap.of("name", PRAVEGA_ID, "namespace", NAMESPACE))
-                .put("spec", ImmutableMap.builder().put("zookeeperUri", zkLocation)
-                                         .put("bookkeeper", bookkeeperSpec)
-                                         .put("pravega", pravegaSpec)
-                                         .build())
+                .put("spec", buildPravegaClusterSpec(zkLocation, bookkeeperSpec, pravegaSpec))
                 .build();
     }
 
-    protected Map<String, Object> getImageSpec(String imageName, String tag) {
-        return ImmutableMap.<String, Object>builder().put("repository", imageName)
-                                                     .put("tag", tag)
-                                                     .put("pullPolicy", IMAGE_PULL_POLICY)
-                                                     .build();
+    /**
+     * As upgrade has been included in operator 0.4.0
+     * a few changes have been included in the Pravega Cluster Spec
+     * like the image tag field for pravega and bookeeper has been replaced by a
+     * single version field which indicates the version for both the pravega and bookeeper components
```

Comment:
Minor: please, use `Pravega` and `Bookkeeper` in capitals, to make it consistent with other places in the documentation/comments.

Reply:
Also, there is a missing dot at the end of the sentence.

Reply:
done